logback/src/main/java/com/grafana/loki/LokiAppender.java

```diff
@@ -0,0 +1,205 @@
+package com.grafana.loki;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.locks.ReentrantLock;
+
+import com.google.protobuf.Timestamp;
+import com.grafana.loki.entry.Entry;
+import com.grafana.loki.logproto.Logproto.Entry.Builder;
+
+import ch.qos.logback.classic.pattern.CallerDataConverter;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.ThrowableProxyUtil;
+import ch.qos.logback.core.UnsynchronizedAppenderBase;
+import ch.qos.logback.core.encoder.EchoEncoder;
+import ch.qos.logback.core.encoder.Encoder;
+
+public class LokiAppender<E> extends UnsynchronizedAppenderBase<E> {
+	
+	protected Encoder<E> encoder = new EchoEncoder<E>();
+
+	protected final ReentrantLock lock = new ReentrantLock(false);
+	
+	protected Map<String, String> additionalFields;
+	
+	protected BlockingQueue<Entry> entryQueue = new LinkedBlockingQueue<Entry>();
+	
+	protected BlockingQueue<Long> ticker = new LinkedBlockingQueue<Long>();
+	
+	public LokiConfig config;
+	
+	private static final String LOGGER = "logback_log_logger";
+	private static final String THREAD = "logback_log_thread";
+	private static final String LEVEL = "logback_log_level";
+	private static final String CALLER = "logback_log_caller";
+	private static final String THROWABLE = "logback_log_throwable";
+	
+	@Override
+    public void start() {
+        super.start();
+        System.out.println("Starting loki logback");
+        Thread tickerThread = new Thread(new Ticker(ticker, config.getBatchWait()*1000));
+        tickerThread.start();
+        Thread th = new Thread(new LokiBatchHandler(entryQueue, ticker, config.getTenantID(), config.getUrl(), config.getBatchWait()*1000, config.getBatchSize()));
+        th.start();
+    }
+	
+	@Override
+    public void stop() {
+        try {
+        	System.out.println("Stopped");
+            super.stop();
+            
+        } finally {
+            try {
+            	
+            } catch (Exception e) {
+                // pass
+            }
+        }
+    }
+	
+	@Override
+	protected void append(E event) {
+		if (event instanceof ILoggingEvent) {
+			ILoggingEvent loggingEvent = (ILoggingEvent) event;
+			handleEvent(loggingEvent);
+		} else {
+			System.out.println("Not IlogginEvent");
```

Comment:
We probably want to get rid of this println?  not sure what anyone would do with it and it could be noisy?

Reply:
Not required. Will remove